plugins/chat/app/services/auto_remove_membership_event_handler.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+
+module Chat
+  module Service
+    # Used to react to various DiscourseEvents from chat/plugin.rb, and
+    # determine which sub-service to call to handle these events. This class
+    # should be called from the [Jobs::AutoJoinChannelMemberships] job,
+    # rather than inline, since some of the operations may take some time.
+    #
+    # We only log the number of users removed based on the original event,
+    # detailed logging has been intentionally left out for now, as admins
+    # should be able to infer what happened to cause user removals from
+    # channels from other staff action logs.
+    #
+    # @example
+    #  Chat::Service::AutoRemoveMembershipEventHandler.call(
+    #    event_type: :chat_allowed_groups_changed,
+    #    event_data: { new_allowed_groups: "1|11" }
+    #  )
+    #
+    class AutoRemoveMembershipEventHandler
```

Comment:
Actually, I don’t think we need this service at all :sweat_smile: 
Since we’re the ones providing the events with their data in `plugin.rb`, we could just call the right service (or job calling the right service) instead. We’ll end up with three jobs instead of one, but this would remove a level of indirection. And it removes the need to call a service from a service :wink:

Reply:
@Flink this is not used now, I am just keeping it for my own reference for the time being with its spec, but will delete it before I send this for proper review.